jujubigdata/handlers.py

```diff
@@ -276,6 +276,28 @@ def restart_namenode(self):
         time.sleep(30)
         self.start_namenode()
 
+    def restart_zookeeper(self):
+        self.stop_zookeeper()
+        time.sleep(5)
+        self.start_zookeeper()
+
+    def stop_zookeeper(self):
+        self._hadoop_daemon('stop', 'zkfc')
+
+    def start_zookeeper(self):
+        self._hadoop_daemon('start', 'zkfc')
+
+    def restart_dfs(self):
+        self.stop_dfs()
+        time.sleep(30)
+        self.start_dfs()
+
+    def stop_dfs(self):
+        self.hadoop_base.run('hdfs', 'sbin/stop-dfs.sh')
```

Comment:
I personally don't think we should have to sleep at all but think there was already a sleep in an existing restart as a 'safety' - 30 sec is probably too long. 5 should be enough if it's needed at all

Andrew M.


> On 20 Apr 2016, at 11:58, Kevin W Monroe <notifications@github.com> wrote:
> 
> In jujubigdata/handlers.py:
> 
> > +        time.sleep(5)
> > +        self.start_zookeeper()
> > +
> > +    def stop_zookeeper(self):
> > +        self._hadoop_daemon('stop', 'zkfc')
> > +
> > +    def start_zookeeper(self):
> > +        self._hadoop_daemon('start', 'zkfc')
> > +
> > +    def restart_dfs(self):
> > +        self.stop_dfs()
> > +        time.sleep(30)
> > +        self.start_dfs()
> > +
> > +    def stop_dfs(self):
> > +        self.hadoop_base.run('hdfs', 'sbin/stop-dfs.sh')
> Why sleep in restart, but not here?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>